pandas/tests/test_indexing.py

```diff
@@ -3511,6 +3511,17 @@ def test_loc_setitem_datetime(self):
             expected = DataFrame({'one' : [100.0, 200.0]},index=[dt1, dt2])
             assert_frame_equal(df, expected)
 
+    def test_loc_setitem_datetimeindex(self):
```

Comment:
pls add the full gamut here (get/setitem) and .loc get/set. also make a loop and test index, list-like, single indexer.
(these all work, but covering bases). 

put these tests near these existing ones on https://github.com/pydata/pandas/blob/master/pandas/tests/test_indexing.py#L856